doc/proposals/composable-integration-in sdk.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Composable integration in SDK
+authors:
+  - "dettori@us.ibm.com"	
+  - "luan@us.ibm.com"
+  - "roytman@il.ibm.com"
+  - "tardieu@us.ibm.com"
+  - "mvaziri@us.ibm.com"
+  
+ 
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-12-16
+last-updated: 2019-12-16
+status: implemented
+see-also:
+replaces:
+superseded-by:
+---
+
+# Composable integration in SDK
+
+## Release Signoff Checklist
+
+- \[x\] Enhancement is `implementable`
+- \[x\] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in [operator-sdk/doc][operator-sdk-doc]
+
+
+## Summary
+
+Kubernetes object specifications often require constant values for their fields. When deploying an entire application
+with many different resources, this limitation often results in the need for staged deployments, because some resources have to be deployed first in order to determine what data to provide for the specifications of dependent resources. This undermines the declarative nature of Kubernetes object specification and requires workflows, manual step-by-step instructions and/or brittle automated scripts for the deployment of applications as a whole.
+
+The Composable SDK can be used to add cross-resource references to any existing CRD, so that values no longer
+need to be hardwired. This feature allows dynamic configuration of a resource, meaning that its fields can be 
+resolved after it has been deployed. 
+
+See this [tutorial](https://github.com/IBM/composable/blob/master/sdk/docs/tutorial.md), in which we add cross-references to the `memcached-operator` using the Composable SDK.
+
+The Composable SDK has been implemented and open-sourced at https://github.com/IBM/composable/tree/master/sdk.
+It is also related to the Composable Operator released on operatorhub.io:
+https://github.com/IBM/composable.
+
+
+## Motivation
+
+To add support for cross-resource references in CRD definitions and controllers (Composable SDK) and make this
+functionality available through the Operator-SDK.
+
```

Comment:
I think it is missing: How it would be added in the SDK?

Reply:
We would add the types that are here:
https://github.com/IBM/composable/blob/master/sdk/composable-types.go

so that operator developers can use the ObjectRef to specify that fields have this type.

We also add the method `ResolveObject` (possibly with a modified interface so that the receiver is the client instead of the first argument) as one more way in which the developer can access objects in etcd (like Get, Update etc...)

Reply:
I think would be nice to have your thoughts on how it should be added to the project too in the doc. Really tks for all.

Reply:
ok will add that. 
Thanks for all the feedback!

Reply:
@camilamacedo86 I changed the proposal doc according to the changes above for the interface.